nodes/stampzilla-server/handlers/securewebsockethandler.go

```diff
@@ -179,6 +180,36 @@ func (wsh *secureWebsocketHandler) Message(s interfaces.MelodySession, msg *mode
 			return err
 		}
 		wsh.WebsocketSender.SendToID(node.UUID, "setup", node)
+	case "setup-device":
+		device := &devices.Device{}
+		err := json.Unmarshal(msg.Body, device)
+		if err != nil {
+			return err
+		}
+
+		logrus.WithFields(logrus.Fields{
+			"from":   msg.FromUUID,
+			"config": device,
+		}).Debug("Received new device configuration")
+
+		node := wsh.Store.GetNode(device.ID.Node)
+		if node == nil {
+			return fmt.Errorf("Node was not found")
+		}
+
+		if node.Aliases == nil {
+			node.Aliases = make(map[devices.ID]string)
+		}
+		node.Aliases[device.ID] = device.Alias
+
+		wsh.Store.AddOrUpdateNode(node)
+		err = wsh.Store.SaveNodes()
```

Comment:
Can we make a SaveNode that only saves one node to reduce risk of 2 people pressing save in the gui overwriting each changes?